docs/design/proposals/clusterid-mapping.md

```diff
@@ -0,0 +1,176 @@
+recreated on the other cluster. When Ceph-CSI receives the request for
+operations like (NodeStage, ExpandVolume, DeleteVolume, etc) the volumeID is
+sent in the request which will help to identify the volume.
+
+```yaml=
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.json: |-
+    [
+      {
+       "clusterID": "rook-ceph",
+       "radosNamespace": "<rados-namespace>",
+       "monitors": [
+         "192.168.39.82:6789"
+       ],
+       "cephFS": {
+         "subvolumeGroup": "<subvolumegroup for cephfs volumes>"
+       }
+      },
+      {
+       "clusterID": "fs-id",
+       "radosNamespace": "<rados-namespace>",
+       "monitors": [
+         "192.168.39.83:6789"
+       ],
+       "cephFS": {
+         "subvolumeGroup": "<subvolumegroup for cephfs volumes>"
+       }
+      }
+    ]
+metadata:
+  name: ceph-csi-config
+```
+
+During CSI/Replication operations, Ceph-CSI will decode the volumeID and gets
+the monitor configuration from the configmap and the poolID will get the
+pool Name and retrieves the OMAP data stored in the rados OMAP and finally
+check the volume is present in the pool.
+
+## Problems with volumeID Replication
+
+* The clusterID can be different
+  * as the clusterID is the namespace where rook is deployed, the Rook might be
+    deployed in the different namespace on a secondary cluster
+  * In standalone Ceph-CSI the clusterID is fsID and fsID is unique per
+    cluster
+
+* The poolID can be different
+  * PoolID which is encoded in the volumeID won't remain the same across
+    clusters
+
+To solve this problem we need to have a new mapping between clusterID's and the
+poolID's.
+
+Example configmap Need to be created before failover to `site2-storage` from
+`site1-storage` and `site3-storage`.
+
+```yaml=
+apiVersion: v1
+kind: ConfigMap
+data:
+  cluster-mapping.json: |-
+  [{
+    "clusterIDMapping": {
+    "site1-storage" (clusterID on site1): "site2-storage" (clusterID on site2)
+   },
+    "RBDPoolIDMapping": [{
+    "1" (poolID on site1): "2" (poolID on site2),
+    "11": "12"
+   }],
+    "CephFSFsIDMapping": [{
+    "13" (FsID on site1): "34" (FsID on site2),
+    "3": "4"
+   }]
+  }, {
+   "clusterIDMapping": {
+   "site3-storage"  (clusterID on site3): "site2-storage" (clusterID on site2)
+   },
+   "RBDPoolIDMapping": [{
+   "5" (poolID on site3): "2" (poolID on site2),
+   "16": "12"
+   }],
+   "CephFSFsIDMapping": [{
+   "3"(FsID on site3): "34" (FsID on site2),
+   "4": "4"
+   }]
+ }]
+metadata:
+  name: ceph-csi-config
+```
+
+**Note:-** the configmap will be mounted as a volume to the CSI (provisioner
+and node plugin) pods.
+
+The above configmap will get created as it is or updated (if new Pools are
+created on the existing cluster) with new entries when the admin choose to
+failover/failback the cluster.
+
+Whenever Ceph-CSI receives a CSI/Replication request it will first decode the
+volumeHandle and try to get the required OMAP details if it is not able to
```

Comment:
I think above sentence need to be rephrased.

Reply:
@humblec please suggest the changes you are thinking of

Reply:
@Madhu-1 after `OMAP details ` you may close that sentence or concatenate it properly

Reply:
@humblec done PTAL